src/Services/Messages/DispatchMessage.php

```diff
@@ -102,14 +103,21 @@ protected function markSent(Message $message, string $messageId): Message
         return $this->markAsSent->handle($message, $messageId);
     }
 
-    /**
-     * Check that the message has not already been sent by getting
-     * a fresh db record
-     */
     protected function isValidMessage(Message $message): bool
     {
-        $message->refresh();
+        $data = $message->newQuery()
```

Comment:
I think it would probably be better/cleaner where to just do something like:

```
$message = $this->messageRepo->find($message->id, ['campaign'])

if ($message->campaign->cancelled) {
  return false;
}

if ((bool)$message->sent_at) {
  return false
}

return true;
```

Reply:
I did this change but have reverted it, as `campaign` isn't a think for messages (they are polymorphic) and it was easier to just leave as-is, working, than work out the best way to manage this correctly with the polymorphic relationship.